services/meta/data.go

```diff
@@ -465,8 +466,33 @@ func (data *Data) DropContinuousQuery(database, name string) error {
 	return ErrContinuousQueryNotFound
 }
 
+// validateURL returns an error if the URL does not have a port or uses a scheme other than UDP or TCP.
+func validateURL(input string) error {
+	u, err := url.Parse(input)
+	if err != nil {
+		fmt.Println(err)
+		return ErrInvalidSubscriptionURL(input)
+	}
+
+	if u.Scheme != "udp" && u.Scheme != "tcp" {
+		return ErrInvalidSubscriptionURL(input)
+	}
+
+	hostParts := strings.Split(u.Host, ":")
+	if len(hostParts) != 2 || hostParts[1] == "" {
+		return ErrInvalidSubscriptionURL(input)
+	}
+	return nil
+}
+
 // CreateSubscription adds a named subscription to a database and retention policy.
 func (data *Data) CreateSubscription(database, rp, name, mode string, destinations []string) error {
+	for _, d := range destinations {
+		if err := validateURL(d); err != nil {
+			return ErrInvalidSubscriptionURL(d)
```

Comment:
`validateURL` only returns an error created with `ErrInvalidSubscriptionURL` so I think you can just `return err` here.